### PR TITLE
Add icon in Data List 

### LINKED ISSUE #1389: Add support on adding an Icon to the DataListItem.
Issue [#1338](https://github.com/patternfly/patternfly-react/issues/1338) was opened in the React repo.  The consumer would like to add an icon to the DataListCell.  The consumer followed the [pattern library](https://www.patternfly.org/pattern-library/content-views/list-view/) and tried to add an icon on the left.  This Pattern Library is PF3 specific and currently PF4 does not support the icon being placed to the left as in PF3.

### PR DESCRIPTION
**Pull request by christiemolloy**
Closes #1389 

Adds a class called `.pf-c-data-list__icon` which styles an icon similar to a checkbox in the data table. Added icon the expandable table example - but it would be good to clarify which examples should use this icon/if any.

### CODE REVIEW

**Comment by mcarrano** [comment]
Can you elaborate on this @christiemolloy ? In not sure what this change is doing. What am I looking for?

**Comment by christiemolloy** [comment]
@mcarrano sorry ill update the description. Take a look at the expandable table with the updated icons.

**Comment by mcarrano** [comment]
Ah, OK I see it @christiemolloy . I was getting confused because the issue says Data List, but then you talk about the expandable table :)  When you did looks fine to me, although I'm still unclear what the purpose of this example is.

File: src/patternfly/components/DataList/data-list.scss [hunk 1]
```diff
@@ -162,6 +171,11 @@ $pf-data-list--breakpoint--mobile: 380px;
   margin-right: var(--pf-c-data-list__check--MarginRight);
 }
 
+.pf-c-data-list__icon {
+  display: flex;
```
Comment (mcoker) [inline src/patternfly/components/DataList/data-list.scss]: is this needed?

**Comment by mcoker** [comment]
@christiemolloy looks great, just one comment.

File: src/patternfly/components/DataList/data-list.scss (hunk 1 above)
Comment (christiemolloy) [reply src/patternfly/components/DataList/data-list.scss]: no, also took it off data-list__check.

**Comment by christiemolloy** [comment]
@mcoker updated. @mcarrano the purpose of adding the icons is to demonstrate that it is possible to have icons in the data list, before this, it was not possible to add an icon and have it style correctly in the data list - refer to the issue #1389

**Pull request merged**